sdk/devcenter/developer-devcenter-rest/test/public/devBoxTests.spec.ts

```diff
@@ -0,0 +1,541 @@
+    expect(pools[0].name).to.equal(poolName);
+  });
+
+  it("GetSchedule", async function () {
+    await createDevBox();
+
+    const scheduleOutput = await client
+      .path(
+        "/projects/{projectName}/pools/{poolName}/schedules/{scheduleName}",
+        projectName,
+        poolName,
+        "default"
+      )
+      .get();
+
+    if (isUnexpected(scheduleOutput)) {
+      throw scheduleOutput.body.error;
+    }
+
+    expect(scheduleOutput.body.name).to.equal("default");
+
+    await deleteDevBox();
+  });
+
+  it("GetSchedules", async function () {
+    await createDevBox();
+
+    const schedulesListResponse = await client
+      .path("/projects/{projectName}/pools/{poolName}/schedules", projectName, poolName)
+      .get();
+
+    if (isUnexpected(schedulesListResponse)) {
+      throw schedulesListResponse.body.error;
+    }
+
+    const schedules: ScheduleOutput[] = [];
+    console.log("Iterating through schedules results:");
+
+    for await (const schedule of paginate(client, schedulesListResponse)) {
+      const { name } = schedule;
+      console.log(`Received schedule "${name}"`);
+      schedules.push(schedule);
+    }
+
+    expect(schedules.length).to.equal(1);
+    expect(schedules[0].name).to.equal("default");
+
+    await deleteDevBox();
+  });
+
+  it("GetDevBox", async function () {
+    await createDevBox();
+
+    const devboxOutput = await client
+      .path(
+        "/projects/{projectName}/users/{userId}/devboxes/{devBoxName}",
+        projectName,
+        userId,
+        devboxName
+      )
+      .get();
+
+    if (isUnexpected(devboxOutput)) {
+      throw devboxOutput.body.error;
+    }
+
+    expect(devboxOutput.body.name).to.equal(devboxName);
+
+    await deleteDevBox();
+  });
+
+  it("GetRemoteConnection", async function () {
+    await createDevBox();
+
+    const remoteConnectionResponse = await client
+      .path(
+        "/projects/{projectName}/users/{userId}/devboxes/{devBoxName}/remoteConnection",
+        projectName,
+        userId,
+        devboxName
+      )
+      .get();
+
+    if (isUnexpected(remoteConnectionResponse)) {
+      throw remoteConnectionResponse.body.error;
+    }
+
+    assert.isTrue(isAValidUrl(remoteConnectionResponse.body.rdpConnectionUrl as string));
+
+    await deleteDevBox();
+  });
+
+  it("GetAllDevBoxes", async function () {
+    await createDevBox();
+
+    const devboxesListResponse = await client.path("/devboxes").get();
+
+    if (isUnexpected(devboxesListResponse)) {
+      throw devboxesListResponse.body.error;
+    }
+
+    const devBoxes: DevBoxOutput[] = [];
+    console.log("Iterating through dev boxes results:");
+
+    for await (const devBox of paginate(client, devboxesListResponse)) {
+      const { name } = devBox;
+      console.log(`Received DevBox "${name}"`);
+      devBoxes.push(devBox);
+    }
+
+    expect(devBoxes.length).to.equal(1);
+    expect(devBoxes[0].name).to.equal(devboxName);
+
+    await deleteDevBox();
```

Comment:
Might be worthwhile to do this as an `afterEach`. Each test using a dev box can set a flag or something, then the `afterEach` hook can `await deleteDevBox();` if the flag is set

Reply:
Fixed: https://github.com/Azure/azure-sdk-for-js/pull/25792/commits/b158c2ea67e700db23ec989b5f5b551f1f084820